packages/mgt-components/src/components/mgt-tasks/mgt-tasks.graph.planner.ts

```diff
@@ -27,110 +29,91 @@ export const addPlannerTask = async (graph: IGraph, newTask: PlannerTask): Promi
+export const removePlannerTask = async (graph: IGraph, task: ITask): Promise<void> => {
   await graph
-    .api(`/planner/tasks/${taskId}`)
+    .api(`/planner/tasks/${task.id}`)
     .header('Cache-Control', 'no-store')
-    .header('If-Match', eTag)
+    .header('If-Match', task.eTag)
     .middlewareOptions(prepScopes('Group.ReadWrite.All'))
     .delete();
 };
 
 /**
  * async promise, allows developer to set a task to complete, associated with taskId
  *
- * @param {string} taskId
- * @param {string} eTag
+ * @param {IGraph} graph
+ * @param {ITask} task
  * @returns {Promise<any>}
  * @memberof Graph
  */
-export const setPlannerTaskComplete = async (graph: IGraph, taskId: string, eTag: string): Promise<void> => {
-  await setPlannerTaskDetails(
-    graph,
-    taskId,
-    {
-      percentComplete: 100
-    },
-    eTag
-  );
+export const setPlannerTaskComplete = async (graph: IGraph, task: ITask): Promise<void> => {
+  await setPlannerTaskDetails(graph, task, { percentComplete: 100 });
 };
 
 /**
  * async promise, allows developer to set a task to incomplete, associated with taskId
  *
- * @param {string} taskId
- * @param {string} eTag
+ * @param {IGraph} graph
+ * @param {ITask} task
  * @returns {Promise<any>}
  * @memberof Graph
  */
-export const setPlannerTaskIncomplete = async (graph: IGraph, taskId: string, eTag: string): Promise<void> => {
-  await setPlannerTaskDetails(
-    graph,
-    taskId,
-    {
-      percentComplete: 0
-    },
-    eTag
-  );
+export const setPlannerTaskIncomplete = async (graph: IGraph, task: ITask): Promise<void> => {
+  await setPlannerTaskDetails(graph, task, { percentComplete: 0 });
 };
 
 /**
  * async promise, allows developer to set details of planner task associated with a taskId
  *
- * @param {string} taskId
- * @param {(PlannerTask)} details
- * @param {string} eTag
+ * @param {IGraph} graph
+ * @param {ITask} task
+ * @param {PlannerTask} details
  * @returns {Promise<any>}
  * @memberof Graph
  */
-export const setPlannerTaskDetails = async (
-  graph: IGraph,
-  taskId: string,
-  details: PlannerTask,
-  eTag: string
-): Promise<any> => {
-  // eslint-disable-next-line @typescript-eslint/no-unsafe-return
-  return await graph
-    .api(`/planner/tasks/${taskId}`)
-    .header('Cache-Control', 'no-store')
-    .middlewareOptions(prepScopes('Group.ReadWrite.All'))
-    .header('If-Match', eTag)
-    .patch(JSON.stringify(details));
+export const setPlannerTaskDetails = async (graph: IGraph, task: ITask, details: PlannerTask): Promise<PlannerTask> => {
+  let response: PlannerTask;
+  try {
+    response = (await graph
+      .api(`/planner/tasks/${task.id}`)
+      .header('Cache-Control', 'no-store')
+      .middlewareOptions(prepScopes('Group.ReadWrite.All'))
+      .header('Prefer', 'return=representation')
+      .header('If-Match', task.eTag)
+      .update(details)) as PlannerTask;
+  } catch (_) {
+    /* empty */
+  }
+  return Promise.resolve(response);
```

Comment:
This doesn't need to be wrapped in a resolved promise here.
Because there was an awaited call in an async method there's already a promise being returned. 

```suggestion
  return response;
```